astroquery/imcce/core.py

```diff
@@ -648,7 +649,17 @@ def _parse_result(self, response, verbose=False):
                     conf.field_names[fieldname]]
 
         # convert object numbers to int
-        results['Number'] = [int(float(n)) for n in results['Number']]
+        # unnumered asteroids return as non numeric values ('-')
+        # this is treated as defaulting to 0, and masking the entry
+        unnumbered_mask = [not str(x).isdigit() for x in results['Number']]
+        numbers = [int(float(x)) if str(x).isdigit()
+                   else 0
+                   for x in results['Number']]
+        asteroid_number_col = MaskedColumn(numbers,
+                                           name='Number',
+                                           mask=unnumbered_mask,
+                                           dtype=np.int)
```

Comment:
nitpicking, but it's better to use `int` rather than the alias. Or in fact drop the whole `dtype` kwarg as `numbers` are already integers.

for relevant discussion see this and the issues linked within: https://github.com/astropy/astropy/pull/6603